code/game/say.dm

```diff
@@ -96,6 +96,15 @@ GLOBAL_LIST_INIT(freqtospan, list(
 	return "[say_mod(input, message_mode)][spanned ? ", \"[spanned]\"" : ""]"
 	// Citadel edit [spanned ? ", \"[spanned]\"" : ""]"
 
+/atom/movable/proc/say_emphasis(input)
+	var/static/regex/italics = regex("\\|(?=\\S)(.*?)(?=\\S)\\|", "g")
+	input = replacetext_char(input, italics, "<i>$1</i>")
+	var/static/regex/bold = regex("\\+(?=\\S)(.*?)(?=\\S)\\+", "g")
+	input = replacetext_char(input, bold, "<b>$1</b>")
+	var/static/regex/underline = regex("_(?=\\S)(.*?)(?=\\S)_", "g")
+	input = replacetext_char(input, underline, "<u>$1</u>")
+	return input
```

Comment:
```suggestion
#define ENCODE_HTML_EPHASIS(input, char, html, varname) \
	var/static/regex/##varname = regex("[char]{2}(.+?)[char]{2}", "g");\
	input = varname.Replace_char(input, "<[html]>$1</[html]>")

/atom/movable/proc/say_emphasis(input)
	ENCODE_HTML_EPHASIS(input, "\\|", "i", italics)
	ENCODE_HTML_EPHASIS(input, "\\+", "b", bold)
	ENCODE_HTML_EPHASIS(input, "_", "u", underline)
	return input

#undef ENCODE_HTML_EPHASIS
```
Tested it this time. Also foolproofed the markdown a little by requiring the markup characters to be inputted twice (ie || italics text ||) to avoid cases such as `1+2+3=6` being rendered as:
1**2**3=6